### DISCUSSION TITLE
Purpose of the distinction between "expense", "revenue" and "transfer"

### DISCUSSION THREAD

**Discussion by Ezwen** [firefly-iii/firefly-iii#8646] [Firefly III] [answered]
Three types of transactions are possible in Firefly: "expense", "revenue" and "transfer".

My question is simple: **what is the purpose of such distinction?** 

The reason for my question is the following: given that a transaction is always from a source account to a destination account, the nature of the transaction could in principle always be derived from said accounts:
- If the source is an "asset account" and the target is an "expense account", then the transaction could be considered as a "**expense**".
- If the source is a "revenue account" and the target it an "asset account", then the transaction could be considered as a "**revenue**".
- If both source and targets are "asset accounts", then the transaction could be considered as a "**transfer**".

Hence, in theory, couldn't Firefly simply contain "transactions", and avoid this (confusing, at least to me) distinction?

**Comment by JC5** [comment, marked as answer]
This is what happens yes, but it's stored regardless to make it easier on the database.

**Reply by Ezwen** [reply]
I see, this is what I suspected. 

Just to understand perfectly: then what exactly happens when one creates a new "expense" and fills the form with a "revenue account" as source and an "asset account" as target? The code ignores that we were about to create an "expense", and instead consider it to be a "revenue"?

If that is the case, then is that distinction really needed UI-wise (regardless of how the database is managed)? Wouldn't a simple _"new transaction"_ button suffice for users? At least for me, this would be less confusing :)

**Reply by JC5** [reply]
The form is already a "new transaction" form, it's the same for each type of transaction. But for most users, this is the less confusing option.

If you press "new transfer" and create a withdrawal, it works.

**Reply by Ezwen** [reply]
Fair enough, if you believe this is the less confusing option, I understand. 

Thank you for taking the time to answer!
